src/stores/api.js

```diff
@@ -23,4 +27,39 @@ export async function getAllTimeGreat(lastThirtyDays=false) {
     raw: true,
   });
   return res.data;
+}
+
+export async function getLoggedInUser() {
+  try {
+    const res = await axios.get(BASE_URL + "/api/user", { raw: true });
+    const username = res.data["username"];
+    localStorage.setItem("currentUser", username);
+  } catch (error) {
+    console.log(error);
+  }
+}
+
+export async function isUserLoggedIn() {
+  const currentUser = localStorage.getItem("currentUser");
+  if (currentUser) {
+    return currentUser;
+  }
+  return;
+}
+
+export async function LogIn() {
+  if (!localStorage.getItem("currentUser")) {
+    await getLoggedInUser();
```

Comment:
More questions!  Why call getLoggedInUser() here?

While Chrome and Edge can handle it, for some reason Firefox doesn't continue running the function after it encounters the axios error here.

Reply:
The flow of this function is as follows
if localStorage does not have currentUser,
call the backend to see if there is a session already
if no session (or above fails)
we go to /api/login


about the browsers, I skipped over that. I will check to confirm the behavior you mention